components/sidebar/sidebar_whats_new_modal/sidebar_whats_new_modal.tsx

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import {Preferences} from 'mattermost-redux/constants';
+import {PreferenceType} from 'mattermost-redux/types/preferences';
+
+import GenericModal from 'components/generic_modal';
+import {ModalIdentifiers} from 'utils/constants';
+
+import './sidebar_whats_new_modal.scss';
+
+type Props = {
+    currentUserId: string;
+    actions: {
+        savePreferences: (userId: string, preferences: PreferenceType[]) => void;
+        closeModal: (modalId: string) => void;
+    };
+}
+
+export default class SidebarWhatsNewModal extends React.PureComponent<Props> {
```

Comment:
Same here, can this be converted to a FC?

Reply:
I don't think so, since @hmhealey had me convert this to be used by the `ModalController`, meaning it needs to be able to handle its own functionality. Again, maybe i'm not understanding FC's properly.

Reply:
Same comment applies.

Reply:
I've never tried using functional components with the `ModalController`, but I don't see any reason it shouldn't work assuming the types are correct for the related actions